plugins/modules/ndo_l3out_interface_group_policy.py

```diff
@@ -0,0 +1,538 @@
+            ),
+            required_together=[
+                ["name", "template"],
+            ],
+        ),
+        custom_qos_policy_uuid=dict(type="str"),
+        state=dict(type="str", default="query", choices=["absent", "query", "present"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "present", ["template", "template_id"], True],
+            ["state", "query", ["template", "template_id"], True],
+            ["state", "absent", ["template", "template_id"], True],
+            ["state", "present", ["l3out", "l3out_uuid"], True],
+            ["state", "query", ["l3out", "l3out_uuid"], True],
+            ["state", "absent", ["l3out", "l3out_uuid"], True],
+            ["state", "absent", ["name"]],
+            ["state", "present", ["name"]],
+        ],
+        mutually_exclusive=[
+            ["interface_routing_policy", "interface_routing_policy_uuid"],
+            ["custom_qos_policy", "custom_qos_policy_uuid"],
+        ],
+    )
+
+    mso = MSOModule(module)
+
+    template_identifier = get_object_identifier(module.params.get("template_id"), module.params.get("template"))
+    l3out_identifier = get_object_identifier(module.params.get("l3out_uuid"), module.params.get("l3out"))
+    name = module.params.get("name")
+    description = module.params.get("description")
+    interface_routing_policy = module.params.get("interface_routing_policy")
+    interface_routing_policy_uuid = module.params.get("interface_routing_policy_uuid")
+    bfd = module.params.get("bfd")
+    bfd_multi_hop = module.params.get("bfd_multi_hop")
+    ospf = module.params.get("ospf")
+    qos_priority = module.params.get("qos_priority")
+    custom_qos_policy = module.params.get("custom_qos_policy")
+    custom_qos_policy_uuid = module.params.get("custom_qos_policy_uuid")
+    state = module.params.get("state")
+
+    mso_template = MSOTemplate(mso, "l3out", template_identifier.get("name"), template_identifier.get("uuid"))
+    mso_template.validate_template("l3out")
+    object_description = "L3Out Interface Group Policy"
+
+    existing_l3outs = mso_template.template.get("l3outTemplate", {}).get("l3outs", [])
+    l3out_object = mso_template.get_object_by_key_value_pairs(
+        "L3Out",
+        existing_l3outs,
+        [KVPair("uuid", l3out_identifier.get("uuid")) if l3out_identifier.get("uuid") else KVPair("name", l3out_identifier.get("name"))],
+        True,
+    )
+    existing_interface_groups = l3out_object.details.get("interfaceGroups", [])
+    if name:
+        match = mso_template.get_object_by_key_value_pairs(
+            object_description,
+            existing_interface_groups,
+            [KVPair("name", name)],
+        )
+        if match:
+            mso.existing = mso.previous = copy.deepcopy(match.details)
+    else:
+        mso.existing = mso.previous = existing_interface_groups
```

Comment:
Insert the `Interface Routing Policy` and `Custom QoS Policy` object names in the `current` and `previous` portion of the module output when we do `create, update, query and delete`.

Reply:
OK